internal/node/tracer.go

```diff
@@ -0,0 +1,80 @@
+package nexnode
+
+import (
+	"context"
+	"log/slog"
+	"time"
+
+	"go.opentelemetry.io/otel"
+	"go.opentelemetry.io/otel/attribute"
+	"go.opentelemetry.io/otel/exporters/otlp/otlptrace"
+	"go.opentelemetry.io/otel/exporters/otlp/otlptrace/otlptracegrpc"
+	"go.opentelemetry.io/otel/propagation"
+	"go.opentelemetry.io/otel/sdk/resource"
+	sdktrace "go.opentelemetry.io/otel/sdk/trace"
+	semconv "go.opentelemetry.io/otel/semconv/v1.21.0"
+	"go.opentelemetry.io/otel/trace"
+	"google.golang.org/grpc"
+	"google.golang.org/grpc/credentials/insecure"
+)
+
+var tracer trace.Tracer
+var tracerProvider *sdktrace.TracerProvider
+
+const (
+	traceServiceName = "nex-node"
+	traceAppName     = "nex"
+)
+
+func InitializeTraceProvider(exporterUrl string) error {
```

Comment:
Should this be tucked under telemetry.go?

Reply:
telemetry and trace are totally separate things and should have their own configs and init. Especially since telemetry is always on and trace exporting is optional.